### PR TITLE
Split up build yml

### PR DESCRIPTION
**Pull request by sindrig**
# Split build.yaml up in 2

Closes #1242 

## What

Faster builds, especially in feature branches

## Why

Because we're losing patience

### CODE REVIEW

**Comment by sindrig** [comment]
image
these are two full builds - still a lot of speed to be gained in the on pull request.

File: .github/workflows/pullrequest.yml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    if: always()
+    needs:
+      - prepare
+      - tests
+      - linting
+      - formatting
+      - build
+    steps:
+      - name: Check tests success
+        run: '[[ ${{ needs.tests.result }} != "failure" ]] || exit 1'
```
Comment (pshomov) [inline .github/workflows/pullrequest.yml:247]: nice, was wondering how to get that.

File: scripts/_chunk.js [hunk 2]
```diff
@@ -16,6 +16,4 @@ let chunks = chunk(
   .map((ch) => ch.join(','))
   .filter((job) => job.length > 0)
 
-if (chunks.length == 0) chunks = ['dummy']
```
Comment (pshomov) [inline scripts/_chunk.js:19]: 👍🏻

**Review by pshomov — APPROVED** [review]

File: .github/workflows/pullrequest.yml (hunk 1 above)
Comment (pshomov) [reply .github/workflows/pullrequest.yml:247]: It is right there in the docs, not that I know about it ;)

**Pull request merged**